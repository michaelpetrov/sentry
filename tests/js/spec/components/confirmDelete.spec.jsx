import React from 'react';
import {mount} from 'enzyme';
import ConfirmDelete from 'app/components/confirmDelete';

describe('ConfirmDelete', function() {
  it('renders', function() {
    let mock = jest.fn();
    let wrapper = mount(
      <ConfirmDelete message="Are you sure?" onConfirm={mock} confirmInput="CoolOrg">
        <button>Confirm?</button>
      </ConfirmDelete>,
      TestStubs.routerContext()
    );
    expect(wrapper.find('Confirm')).toMatchSnapshot();
  });

  it('confirm button is disabled and bypass prop is false when modal opens', function() {
    let mock = jest.fn();
    let wrapper = mount(
      <ConfirmDelete message="Are you sure?" onConfirm={mock} confirmInput="CoolOrg">
        <button>Confirm?</button>
      </ConfirmDelete>,
      TestStubs.routerContext()
    );

    wrapper.find('button').simulate('click');

    expect(wrapper.find('Confirm').prop('bypass')).toBe(false);
    expect(wrapper.state('disableConfirmButton')).toBe(true);
  });

  it('confirm button stays disabled with non-matching input', function() {
    let mock = jest.fn();
    let wrapper = mount(
      <ConfirmDelete message="Are you sure?" onConfirm={mock} confirmInput="CoolOrg">
        <button>Confirm?</button>
      </ConfirmDelete>,
      TestStubs.routerContext()
    );
    wrapper.find('button').simulate('click');
    // simulating handleChange()
    wrapper.setState({confirmInput: 'Cool', disableConfirmButton: true});
    expect(wrapper.find('Confirm').prop('disableConfirmButton')).toBe(true);
  });

  it('confirm button is enabled when confirm input matches', function() {
    let mock = jest.fn();
    let wrapper = mount(
      <ConfirmDelete message="Are you sure?" onConfirm={mock} confirmInput="CoolOrg">
        <button>Confirm?</button>
      </ConfirmDelete>,
      TestStubs.routerContext()
    );
    wrapper.find('button').simulate('click');
    // simulating handleChange()
    wrapper.setState({confirmInput: 'CoolOrg', disableConfirmButton: false});
    expect(wrapper.find('Confirm').prop('disableConfirmButton')).toBe(false);

    wrapper
      .find('Button')
      .last()
      .simulate('click');

    expect(
      wrapper
        .find('Modal')
        .first()
        .prop('show')
    ).toBe(false);
    expect(mock).toHaveBeenCalled();
    expect(mock.mock.calls).toHaveLength(1);
  });
});
