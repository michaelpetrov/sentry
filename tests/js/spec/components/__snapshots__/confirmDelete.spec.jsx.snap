// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDelete renders 1`] = `
<Confirm
  bypass={false}
  cancelText="Cancel"
  confirmText="Confirm"
  disableConfirmButton={true}
  message={
    <UNDEFINED>
      <Alert
        type="error"
      >
        Are you sure?
      </Alert>
      <Field
        alignRight={false}
        disabled={false}
        flexibleControlStateSize={true}
        inline={false}
        label={
          <div>
            Please enter 
            <code>
              CoolOrg
            </code>
             to confirm the deletion
          </div>
        }
        p={0}
        required={false}
        visible={true}
      >
        <Input
          onChange={[Function]}
          placeholder="CoolOrg"
          type="text"
          value=""
        />
      </Field>
    </UNDEFINED>
  }
  onConfirm={[MockFunction]}
  priority="primary"
>
  <button
    onClick={[Function]}
  >
    Confirm?
  </button>
  <Modal
    animation={false}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropClassName="modal-backdrop"
      containerClassName="modal-open"
      enforceFocus={true}
      keyboard={true}
      manager={
        ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        }
      }
      onEntering={[Function]}
      onExited={[Function]}
      onHide={[Function]}
      renderBackdrop={[Function]}
      restoreFocus={true}
      show={false}
    />
  </Modal>
</Confirm>
`;
